platform/solr/solr-query/src/main/java/org/codice/solr/query/SolrQueryFilterVisitor.java

```diff
@@ -68,39 +85,106 @@ public SolrQueryFilterVisitor(SolrClient client, String solrCoreName) {
         schemaFieldResolver = new SchemaFieldResolver(client);
         this.solrCoreName = solrCoreName;
     }
+    @Override
+    public Object visit(And filter, Object data) {
+        List<Filter> childList = filter.getChildren();
+        return logicalOperator(childList, AND, data);
+    }
+
+    @Override
+    public Object visit(Or filter, Object data) {
+        List<Filter> childList = filter.getChildren();
+        return logicalOperator(childList, OR, data);
+    }
+
+    private Object logicalOperator(List<Filter> filters, String operator, Object data) {
+        if (filters == null || filters.isEmpty()) {
+            throw new UnsupportedOperationException(
+                    "[" + operator + "] operation must contain 1 or more filters.");
+        }
+        int startIndex = 0;
+        Filter child = filters.get(startIndex);
+        startIndex++;
+        StringBuilder builder = new StringBuilder();
+        builder.append(START_PAREN);
+        builder.append(getQuery(child.accept(this, data)));
+
+        for (int i = startIndex; i < filters.size(); i++) {
+            Filter localFilter = filters.get(i);
+
+            if (localFilter != null) {
+                String query = getQuery(localFilter.accept(this, data));
+                if(query == null){
+                    throw new UnsupportedOperationException(
+                            "Query operation "+localFilter+" is not supported.");
+                }
+                builder.append(operator)
+                        .append(query);
+            } else {
+                throw new UnsupportedOperationException(
+                        "Query was not interpreted properly. Query should not be null.");
+            }
+
+        }
+        builder.append(END_PAREN);
+
+        return new SolrQuery(builder.toString());
+    }
+
+    private String getQuery(Object data) {
+        if (data instanceof SolrQuery) {
+            return ((SolrQuery) data).getQuery();
+        }
+        return null;
+    }
 
     @Override
     public Object visit(PropertyIsEqualTo filter, Object data) {
         LOGGER.debug("ENTERING: PropertyIsEqualTo filter");
 
         ExpressionValueVisitor expressionVisitor = new ExpressionValueVisitor();
 
-        filter.getExpression1()
+        String propertyName = (String) filter.getExpression1()
                 .accept(expressionVisitor, data);
-        filter.getExpression2()
+        String literalValue = (String) filter.getExpression2()
                 .accept(expressionVisitor, data);
 
-        String propertyName = expressionVisitor.getPropertyName();
+        String mappedPropertyName = getMappedPropertyName(propertyName);
```

Comment:
❓ Don't we still need to make sure `propertyName` and `literalValue` are not blank?